### PR TITLE
Fix/rhut/animations transitions

### PR DESCRIPTION
**Pull request by Rhut-virani**
- Animate Gradient Transitions via the scroll position and sunset vs sunrise time.

### CODE REVIEW

File: src/app/components/AnimatedBackground.tsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  }
+
+  // Convert hex to RGB to allow interpolation
+  const hexToRgb = (hex: string) => {
+    const bigint = parseInt(hex.substring(1), 16)
+    const r = (bigint >> 16) & 255
+    const g = (bigint >> 8) & 255
+    const b = bigint & 255
+    return [r, g, b]
+  }
```
Comment (theolvq) [inline src/app/components/AnimatedBackground.tsx:136]: Fancy! Bitwise operators???

File: src/app/components/AnimatedBackground.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+      )
+    if (JSON.stringify(currentColors) !== JSON.stringify(nextColors)) {
+      setscrollGradientPosition(timeStampDifference)
+    }
+    updateColors()
+  }, [timestamp?.graphTimeStamp])
+
+  return (
+    <div
+      className={`${card ? 'h-full max-h-[150px]' : 'fixed min-h-[100vh] '} inset-0 -z-10`}
```
Comment (theolvq) [inline src/app/components/AnimatedBackground.tsx:189]: Let's use clsx for that stuff please, it keeps things cleaner

File: src/app/components/AnimatedBackground.tsx
Comment (theolvq) [inline src/app/components/AnimatedBackground.tsx:1]: Pretty cool stuff going on in there

File: src/app/components/AnimatedBackground.tsx (hunk 1 above)
Comment (theolvq) [reply src/app/components/AnimatedBackground.tsx:136]: Let's explain what's going on here, I personally have no idea what those operators do.

**Review by theolvq — APPROVED** [review]
Looks good! Except that runtime error

File: src/app/components/AnimatedBackground.tsx (hunk 1 above)
Comment (Rhut-virani) [reply src/app/components/AnimatedBackground.tsx:138]: Its all Maia's work. 
But I can add some notes.

Comment (Rhut-virani) [reply src/app/components/AnimatedBackground.tsx:138]: I found this solution online. 
Will add the note here.

**Review by theolvq — APPROVED** [review]
The fog bug is fixed, thank you!
The scroll feels a little bit "choppy" for some reason when clicking on a day. Here's a quick loom to explain better: https://www.loom.com/share/30ab90f2609e4e0291045adf35606e3a?sid=a69b95ad-337d-41f7-98df-9834f692e867
Thanks for adding explanations on the bitwise stuff, very helpful

**Pull request merged**